test/CMakeLists.txt

```diff
@@ -54,6 +54,69 @@ add_executable(${test_target} ${SPARROW_TESTS_SOURCES})
 target_link_libraries(${test_target} PRIVATE sparrow doctest::doctest)
 add_test(NAME ${test_target} COMMAND ${test_target})
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    target_compile_options(${test_target} 
+        PRIVATE
+            /permissive-
+            /W4
+            /w14640
+            /external:anglebrackets
+            /external:W0)
+elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    target_compile_options(${test_target}
+        PRIVATE 
+            -Wall
+            -Wextra
+            -pedantic
+            -Werror
+            -Wno-unused-parameter
+            -Wno-unused-variable
+            -Wno-unused-function
+            -Wshadow
+            -Wnon-virtual-dtor
+            -Wold-style-cast
+            -Wcast-align
+            -Wunused
+            -Woverloaded-virtual
+            -Wpedantic
+            -Wconversion
+            -Wsign-conversion
+            -Wmisleading-indentation
+            -Wduplicated-cond
+            -Wduplicated-branches
+            -Wlogical-op
+            -Wnull-dereference
+            -Wuseless-cast
+            -Wdouble-promotion
+            -Wformat=2 
+            -Wimplicit-fallthrough
```

Comment:
Naive question: do you think it would be worth factorizing flags used by  both `gcc` and `clang`?

Reply:
How about defining a list for the common ones?

Not that this suggestion need not be addressed at all.